crypto/dilithium/p_pqdsa_test.cc

```diff
@@ -1343,42 +1434,97 @@ TEST_P(PQDSAParameterTest, SIGOperations) {
       0x4a, 0x41, 0x4b, 0x45, 0x20, 0x4d, 0x41, 0x53, 0x53, 0x49,
       0x4d, 0x4f, 0x20, 0x41, 0x57, 0x53, 0x32, 0x30, 0x32, 0x31, 0x2e};
 
-  ASSERT_TRUE(EVP_DigestSignInit(md_ctx.get(), nullptr, nullptr,
-                                 nullptr, pkey.get()));
-  ASSERT_TRUE(EVP_DigestSign(md_ctx.get(), signature1.data(), &sig_len,
-                             msg1.data(), msg1.size()));
+  // ---- 2. Test signature round trip (sign + verify) ----
+
+  // Initalize the signing context |md_ctx| with the |pkey| we generated
+  ASSERT_TRUE(EVP_DigestSignInit(md_ctx.get(), nullptr, nullptr, nullptr, pkey.get()));
+
+  // To sign, we first need to allocate memory for the signature. We call
+  // EVP_DigestSign with sig = nullptr to indicate that we are doing a size check
+  // on the signature size. The variable |sig_len| will be returned with the
+  // correct signature size, so we can allocate memory.
+  size_t sig_len = 0;
+  ASSERT_TRUE(EVP_DigestSign(md_ctx.get(), nullptr, &sig_len, msg1.data(), msg1.size()));
+
+  // Verify that the returned signature size is as expected
+  ASSERT_EQ(sig_len, GetParam().signature_len);
+
+  // Allocate memory for the signature and sign first message; message1
+  std::vector<uint8_t> sig1(sig_len);
+  ASSERT_TRUE(EVP_DigestSign(md_ctx.get(), sig1.data(), &sig_len, msg1.data(), msg1.size()));
 
   // Verify the correct signed message
-  ASSERT_TRUE(EVP_DigestVerify(md_ctx.get(), signature1.data(), sig_len,
-                               msg1.data(), msg1.size()));
+  ASSERT_TRUE(EVP_DigestVerifyInit(md_ctx_verify.get(), nullptr, nullptr, nullptr, pkey.get()));
+  ASSERT_TRUE(EVP_DigestVerify(md_ctx_verify.get(), sig1.data(), sig_len, msg1.data(), msg1.size()));
 
-  // Verify the signed message fails upon a different message
-  ASSERT_FALSE(EVP_DigestVerify(md_ctx.get(), signature1.data(), sig_len,
-                                msg2.data(), msg2.size()));
+  // ---- 3. Test signature failure modes: incompatible messages/signatures ----
+  // Check that the verification of signature1 failes for a different message; message2
+  ASSERT_FALSE(EVP_DigestVerify(md_ctx_verify.get(), sig1.data(), sig_len, msg2.data(), msg2.size()));
+  uint32_t err = ERR_get_error();
+  EXPECT_EQ(ERR_LIB_EVP, ERR_GET_LIB(err));
+  EXPECT_EQ(EVP_R_INVALID_SIGNATURE, ERR_GET_REASON(err));
 
-  // Sign the different message
+  // reset the contexts between tests
+  md_ctx.Reset();
+  md_ctx_verify.Reset();
+
+  // PQDSA signature schemes can be in either randomized (every signature on a
```

Comment:
```suggestion
  // PQDSA signature schemes can be either in randomized (every signature on a
```

Reply:
fixed in [f7d1dad](https://github.com/aws/aws-lc/pull/2062/commits/f7d1dadb64f54b7695d871a88d4dfacf42048637)